Subscriber/ApplePayDirectSubscriber.php

```diff
@@ -26,9 +65,62 @@ public static function getSubscribedEvents()
             'Enlight_Controller_Action_PostDispatch_Frontend' => 'onFrontendPostDispatch',
             'Enlight_Controller_Action_PostDispatch_Widget' => 'onFrontendPostDispatch',
             'Enlight_Controller_Action_PostDispatch_Frontend_Checkout' => 'onFrontendCheckoutPostDispatch',
+            'Enlight_Controller_Action_PreDispatch_Frontend_Checkout' => 'onFrontendCheckoutPreDispatch',
+            'Enlight_Controller_Action_PostDispatch_Frontend_Account' => 'onFrontendAccountPostDispatch',
         ];
     }
 
+    /**
+     * @param Enlight_Event_EventArgs $args
+     */
+    public function onFrontendAccountPostDispatch(\Enlight_Event_EventArgs $args)
+    {
+        /** @var Enlight_Controller_Request_RequestHttp $request */
+        $request = $args->get('request');
+
+        if ($request->getActionName() !== 'login') {
+            return;
+        }
+
+        if ($this->modules->Admin()->sCheckUser() === false) {
+            return;
+        }
+
+        $currentPaymentMethod = $this->modules->Admin()->sGetUserData()['additional']['payment']['name'];
+
+        if ($this->paymentMethodService->isApplePayPaymentMethod($currentPaymentMethod) === false) {
+            return;
+        }
+
+        $userId = (int)$this->modules->Admin()->sGetUserData()['additional']['user']['id'];
+        $paymentId = $this->accountService->getCustomerDefaultNonApplePayPaymentMethod($userId);
+
+        $this->accountService->updateCustomerDefaultPaymentMethod($userId, $paymentId);
+    }
+
+    /**
+     * @param Enlight_Event_EventArgs $args
+     */
+    public function onFrontendCheckoutPreDispatch(\Enlight_Event_EventArgs $args)
+    {
+        /** @var Enlight_Controller_Request_RequestHttp $request */
+        $request = $args->get('request');
+
+        if ($request->getActionName() !== 'finish') {
+            return;
+        }
+
+        $currentPaymentMethod = $this->modules->Admin()->sGetUserData()['additional']['payment']['name'];
```

Comment:
just something that popped up in my mind
if this code breaks, will the checkout also break?
if so we would need a try/catch and a logger with a warning...because that feature must never break a checkout

Reply:
This will always be set when calling sGetuserData:

`            $userData['additional']['payment'] = $this->sGetPaymentMeanById(
                $userData['additional']['user']['paymentID'],
                $userData
            );`

Reply:
i was talking about this whole code and process...so if something breaks when updating the payment method

Reply:
I've added try catch just to be sure

Reply:
thank you